esmvalcore/_provenance.py

```diff
@@ -15,13 +15,6 @@
 ESMVALTOOL_URI_PREFIX = 'https://www.esmvaltool.org/'
 
 
-def update_without_duplicating(bundle, other):
-    """Add new records from other provenance bundle."""
-    new_records = set(other.records) - set(bundle.records)
-    for record in new_records:
-        bundle.add_record(record)
-
-
```

Comment:
Why is this not needed anymore? Was it ever?

Reply:
It was needed to avoid duplicate provenance items. Removing the duplicates is now only done once, just before saving the provenance. This is much faster than doing it every time there is an update.

Reply:
Duplicate records appear when merging the provenance of two files, e.g. they might share some ancestor files. For example, if you plot the multi model mean and median in the same figure, all input files would be listed twice in the provenance.